### PR TITLE
Replace sun_intensity with sun irradiance

### PR DESCRIPTION
**Pull request by Hirmuolio**
#### Summary
Features "Replace sun_intensity with sun irradiance"

#### Purpose of change

`sun_intensity` is used to measure intnesity of sunlight. It is not very good at doing that since it does not check the sunlight.

For example you'd get sunlight warmth early in the morning before sun was even above horizon. And sunlight warmth on sunny days was same on mid summer and mid winter days.

#### Describe the solution

Use sun irradiance instead of `sun_intensity `.

The values used are mostly arbitrary. The old values were even more arbitrary so it is not easy to pick "correct" numbers. The values of the numbers are very open for suggestions.

Some numbers are especially arbitrary:

Sunlight warmth for body used to be 1000 on sunny days and 500 on other days. Now it is 1.5*sun irradiance (irradiance at midday during summer is about 1000).

Weather factor for leaf mutagens used to be 1 in "normal" sunlight and 0.5 otherwise. Now it is 1 in "irradiance::moderate" sunlight and irradiance/irradiance::moderate at below that irradiance.

Other affected things: Troglodyte and light sensitivity mutations, cloromorph mutation, leaf mutation, ligthing fire with lenses, glare.

Also fixed minor bug where irradiance was negative in between nautical dawn and sunrise.

#### Describe alternatives you've considered

Various things that use light level for things could instead use irradiance.

#### Testing

Tests pass.

### CODE REVIEW

**Comment by PatrikLundell** [comment]
I don't understand the alternatives comment. Light level is presumably how bright it is at a location, while "irradiance" presumably is the light expressively provided by the sun. They should presumably be the same outdoors outside of shadows, but indoors and in shadows (which I assume are not implemented) the value should be different, and it should be different when other light sources are in play as well (including a full moon). I'm not even sure outdoors during the day should use it exclusively either, as there is the possibility of artificial lighting, in particular search and flood lights that ought to provide more light than the sun during a cloudy winter evening at the very least.

**Comment by Hirmuolio** [comment]
Light level is somewhat arbitrary approximation for how well you see in the illumination. 0 for zero visibility and 100 for full visibility. (higher than 100 values are used in some places but those can be mostly ignored).

Irradiance is the "real" thing in W/m2.

For irradiance vs light level the biggest point is that you have "full vision illumination" in pretty low illumination (relatively speaking).

As soon as sun is above horizon you can see perfectly fine (actually a bit before). But obviously the irradiance from Sun is pretty low at that point.

There aren't many places where light level is used for non vision things. One example is how long it takes to light fire with magnifying glass depends on the light level. It would be better to scale it on irradiance.

**Comment by PatrikLundell** [comment]
I don't have an issue with translating irradiance into light level from the solar light source (and agree that magnifying glass fire lighting ought to be base on the power received from the sun: it won't do much in a room brightly lit from light tubes in the ceiling). However, there's a need to calculate a light level that both takes into account that the irradiance essentially tops out fairly early, and that there potentially are other light sources out there.

File: tests/behavior_test.cpp [hunk 1]
```diff
@@ -155,6 +155,7 @@ TEST_CASE( "behavior_tree", "[behavior]" )
 TEST_CASE( "check_npc_behavior_tree", "[npc][behavior]" )
 {
     clear_map();
+    calendar::turn = calendar::start_of_cataclysm;
```
Comment (Fris0uman) [inline tests/behavior_test.cpp:158]: Why this line? this seems unrelated to the rest of the PR

Comment (Hirmuolio) [reply tests/behavior_test.cpp:158]: Warmth from sun is now affected by weather and time of day in wider range of situations.

That is there to set time to middle of night to prevent warmth from sun from messing up the results.
